### ISSUE TITLE
Cannot set environmental variables for custom service

### ISSUE THREAD

**Issue by rschroll** [jupyterhub/zero-to-jupyterhub-k8s#377]
I'm trying to include a custom service in the configuration yaml file.  Assuming I understand correctly, this should be a valid yaml segment to start a particular service:
```
hub:
  services:
    hashauth:
      admin: true
      url: 'http://127.0.0.1:10101'
      command: ['hashauthservice']
      environment: {'CONFIG_FILE': '/srv/jupyterhub_config.py'}
```

However, when this is pushed to a cluster, the hub fails to launch, with the error `traitlets.traitlets.TraitError: The 'environment' trait of a Service instance must be a dict, but a value of class 'str' (i.e. 'map[CONFIG_FILE:/srv/jupyterhub_config.py]') was specified.`

My guess is that, in one of the transformations of the configs files, this attribute wasn't correctly deserialized back into a dictionary.  The full output is below, but it's a bit confused.  Presumably stdout and stderr are getting comingled.

```
[I 2018-01-04 23:21:20.756 JupyterHub app:830] Loading cookie_secret from env[JPY_COOKIE_SECRET]
[I 2018-01-04 23:21:20.806 JupyterHub app:983] Not using whitelist. Any authenticated user will be allowed.
[E 2018-01-04 23:21:20.832 JupyterHub app:1623]
    Traceback (most recent call last):
      File "/usr/local/lib/python3.5/dist-packages/jupyterhub/app.py", line 1620, in launch_instance_async
        yield self.initialize(argv)
      File "/usr/lib/python3.5/types.py", line 177, in send
        return self.__wrapped.send(val)
      File "/usr/local/lib/python3.5/dist-packages/jupyterhub/app.py", line 1379, in initialize
        self.init_services()
      File "/usr/local/lib/python3.5/dist-packages/jupyterhub/app.py", line 1126, in init_services
        setattr(service, key, value)
      File "/usr/local/lib/python3.5/dist-packages/traitlets/traitlets.py", line 585, in __set__
        self.set(obj, value)
      File "/usr/local/lib/python3.5/dist-packages/traitlets/traitlets.py", line 559, in set
        new_value = self._validate(obj, value)
      File "/usr/local/lib/python3.5/dist-packages/traitlets/traitlets.py", line 591, in _validate
        value = self.validate(obj, value)
      File "/usr/local/lib/python3.5/dist-packages/traitlets/traitlets.py", line 2529, in validate
hub.concurrent-spawn-limit 64
singleuser.start-timeout 300
singleuser.uid 1000
singleuser.fs-gid 1000
singleuser.node-selector {}
singleuser.storage.type dynamic
singleuser.storage.capacity 10Gi
singleuser.storage.home_mount_path /home/vagrant
singleuser.storage.extra-volumes []
singleuser.storage.extra-volume-mounts []
singleuser.lifecycle-hooks {'postStart': {'exec': {'command': ['/bin/bash', '/srv/copy.sh']}}}
singleuser.memory.guarantee 2G
auth.type custom
auth.custom.class-name hashauthenticator.HashAuthenticator
auth.custom.config {'secret_key': <redacted>}
singleuser.extra-env {<redacted, but valid>}
auth.admin.access True
cull.enabled True
cull.timeout 5400
cull.every 600
        value = super(Dict, self).validate(obj, value)
cull.users False
      File "/usr/local/lib/python3.5/dist-packages/traitlets/traitlets.py", line 1677, in validate
        self.error(obj, value)
hub.services {'hashauth': {'environment': 'map[CONFIG_FILE:/srv/jupyterhub_config.py]', 'url': 'http://127.0.0.1:10101', 'admin': True, 'command': ['hashauthservice']}}
      File "/usr/local/lib/python3.5/dist-packages/traitlets/traitlets.py", line 1524, in error
hub.base_url /
        raise TraitError(e)
    traitlets.traitlets.TraitError: The 'environment' trait of a Service instance must be a dict, but a value of class 'str' (i.e. 'map[CONFIG_FILE:/srv/jupyterhub_config.py]') was specified.
    
hub.db_url sqlite:///jupyterhub.sqlite
singleuser.cmd jupyterhub-singleuser
```

**Comment by yuvipanda** [comment]
Thank you for the detailed report, @rschroll! Do you have the ability to test out https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/385 to see if that solves your problem?

**Comment by minrk** [comment]
Closed by #385
